athenahealth/httpclient.go

```diff
@@ -340,6 +346,18 @@ func (h *HTTPClient) WithStats(stats Stats) *HTTPClient {
 	return h
 }
 
+func (h *HTTPClient) WithTracer(tracer Tracer) *HTTPClient {
+	h.tracer = tracer
+
+	return h
+}
+
+func (h *HTTPClient) WithDatadogTracer(opts ...tracer.Option) *HTTPClient {
```

Comment:
When would you use WithDatadogTracer over WithTracer? Wouldn't tracer options be configured on the tracer itself?

Reply:
If we wanted to implement the tracer in calling code that could make sense but using `WithDatadogTracer` allows the caller to not need to implement before and after methods.
I also just realized that there's some nice helper methods in `gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http` to wrap the http client transport which automatically hooks into before and after. This may be a preferred option.